File: structural/proxy/typescript/service.ts
import { IUserService } from './interface';

export class UserService implements IUserService {
	async getUser(id: number): Promise<string> {
		console.log(`Fetching user with id ${id} from the API...`);
		return new Promise(resolve => {
			setTimeout(() => resolve(`User ${id}`), 1000);
		});
	}
}

File: structural/proxy/python/service.py
import time

from interface import UserServiceModel


class UserService(UserServiceModel):
    def get_user(self, user_id: int) -> str:
        print(f"Fetching user with id {user_id} from the API...")
        time.sleep(1)  # Simulate a network call
        return f"User {user_id}"
